backend/src/main/repositories/NotificationRepository.ts

```diff
@@ -0,0 +1,134 @@
+    };
+
+    public delete = async (id: number): Promise<number> => {
+        try {
+            const deletedNotificationStatus = await Notification.destroy({
+                where: {
+                    id: id
+                }
+            });
+
+            log(`Notification with id ${id} has been deleted`);
+            return Promise.resolve(deletedNotificationStatus);
+        } catch (err: any) {
+            log(err);
+            return Promise.resolve(err);
+        }
+    };
+
+    public update = async (id: number, updatedValue: NotificationUpdateDTO): Promise<number> => {
+        try {
+            await Notification.update(updatedValue, {
+                where: {
+                    id: id
+                }
+            });
+
+            log(`Notification with id ${id} has been updated`);
+            return Promise.resolve(1);
+        } catch (err: any) {
+            return Promise.reject(err);
+        }
+    };
+
+    public get = async (specificNotification: { id: number, email: string }): Promise<Notification | null> => {
+        try {
+            const notification = await Notification.findOne({
+                where: {
+                    id: specificNotification.id,
+                    email: specificNotification.email
+                }
+            });
+
+            log(`Notification with id ${notification?.id} and email ${notification?.email} has been retrieved`);
+            return Promise.resolve(notification);
+        } catch (err: any) {
+            log(err);
+            return Promise.reject(err);
+        }
+    };
+
+    public getByEmail = async (email: string): Promise<Notification | null> => {
+        try {
+            const notification = await Notification.findOne({
+                where: {
+                    email: email
+                }
+            });
+
+            log(`Notification with id ${notification?.id} and email ${notification?.email} has been retrieved`);
+            return Promise.resolve(notification);
+        } catch (err: any) {
+            log(err);
+            return Promise.reject(err);
+        }
+    };
+
+    public getById = async (id: number): Promise<Notification | null> => {
+        try {
+            const notification = await Notification.findOne({
+                where: {
+                    id: id
+                }
+            });
+
+            log(`Notification with id ${notification?.id} and email ${notification?.email} has been retrieved`);
+            return Promise.resolve(notification);
+        } catch (err: any) {
+            log(err);
+            return Promise.reject(err);
+        }
+    };
+
+    public getAllByEmail = async (email: string): Promise<Notification[] | null> => {
+        try {
+            const notification: Notification[] = await Notification.findAll({
+                where: {
+                    email: email
+                }
+            });
+
+            log(`${notification} has been retrieved`);
```

Comment:
similar to the previous comment. Printing the array might not be beautiful to read.

Reply:
Reworded the log message